pkg/dynamicinstrumentation/codegen/c/expressions.h

```diff
@@ -0,0 +1,309 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+#ifndef DI_EXPRESIONS_H
+#define DI_EXPRESIONS_H
+
```

Comment:
@val06 This is the place I'd recommend starting for reading through the code.

These functions are the implementations of individual operations that all bpf captures string together. They're generated in the function `GenerateLocationExpression()` (in location_expression.go) as LocationExpression{} structs. Those structs are put in place for the target elements we're capturing. They're later collected in order (`collectLocationExpressions()`) and then generated using the templates defined in `expression_templates.go`.

Reply:
please add a comment for each one of those building blocks functions to explain shortly its purpose (providing the context of DI use-case)

also document the input parameters when appropriate